tools/bin/check_images_exist.sh

```diff
@@ -82,16 +82,18 @@ checkPlatformImages() {
 
 checkNormalizationImages() {
   echo -e "$blue_text""Checking Normalization images exist...""$default_text"
-  # the only way to know what version of normalization the platform is using is looking in NormalizationRunnerFactory.
   local image_version;
-  factory_path=airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/NormalizationRunnerFactory.java
+  definition_file_path=airbyte-config/init/src/main/resources/seed/destination_definitions.yaml
   # -f True if file exists and is a regular file
-  if ! test -f $factory_path; then
-    echo -e "$red_text""No NormalizationRunnerFactory found at path! H4LP!!!""$default_text"
+  if ! test -f $definition_file_path; then
+    echo -e "$red_text""Destination definition file not found at path! H4LP!!!""$default_text"
   fi
-  image_version=$(cat $factory_path | grep 'NORMALIZATION_VERSION =' | cut -d"=" -f2 | sed 's:;::' | sed -e 's:"::g' | sed -e 's:[[:space:]]::g')
-  echo -e "$blue_text""Checking normalization images with version $image_version exist...""$default_text"
-  VERSION=$image_version 
+  normalization_image_versions=$(cat $definition_file_path | grep 'normalizationTag:' | cut -d":" -f2 | sed 's:;::' | sed -e 's:"::g' | sed -e 's:[[:space:]]::g')
+  IFS=' ' read -r -a array <<< "$normalization_image_versions"
```

Comment:
Since we have normalization image and tag as part of the definition file, I feel like this check can be further simplified to be more like the `checkConnectorImages` function, looking up the normalization image and normalization tag pair to see if it exists on docker hub. The current aproach requires a docker-compose file only to have the image names, which doesn't seem necessary anymore. Feel free to make a separate issue / different PR for updating this script though.

Reply:
This is a good point. I agree with you, but I suggest leaving these changes, and in the next pull request update them as you described. Because the script will not find the required class and part of the checks will not pass.